project/src/main/kotlin/org/ionproject/core/writeApi/insertClassSectionEvents/InsertClassSectionEventsRepoImpl.kt

```diff
@@ -0,0 +1,106 @@
+import org.ionproject.core.writeApi.insertClassSectionEvents.sql.InsertClassSectionEventsData.SCHOOL_NAME_PARAM
+import org.ionproject.core.writeApi.insertClassSectionEvents.sql.InsertClassSectionEventsData.SUMMARY_PARAM
+import org.ionproject.core.writeApi.insertClassSectionEvents.sql.InsertClassSectionEventsData.WEEK_DAYS_PARAM
+import org.jdbi.v3.core.transaction.TransactionIsolationLevel
+import org.springframework.stereotype.Component
+import java.sql.Timestamp
+
+@Component
+class InsertClassSectionEventsRepoImpl(
+  private val tm: TransactionManager
+) : InsertClassSectionEventsRepo {
+
+  override fun insertClassSectionSchoolInfo(
+    schoolName: String?,
+    schoolAcr: String?,
+    programmeName: String?,
+    programmeAcr: String?,
+    programmeTermSize: Int?,
+    calendarTerm: String?
+  ): Unit = tm.run(TransactionIsolationLevel.SERIALIZABLE) { handle ->
+    handle
+      .createCall(CALL_UPSERT_SCHOOL)
+      .bind(SCHOOL_NAME_PARAM, schoolName)
+      .bind(SCHOOL_ACR_PARAM, schoolAcr)
+      .bind(PROGRAMME_NAME_PARAM, programmeName)
+      .bind(PROGRAMME_ACR_PARAM, programmeAcr)
+      .bind(PROGRAMME_TERM_SIZE_PARAM, programmeTermSize)
+      .bind(CALENDAR_TERM_PARAM, calendarTerm)
+      .invoke()
+  }
+
+  override fun insertClassSectionCourseInfo(
+    courseName: String?,
+    courseAcr: String?,
+    calendarSection: String?,
+    calendarTerm: String?,
+    language: String,
+    category: String
+  ): Unit = tm.run(TransactionIsolationLevel.SERIALIZABLE) { handle ->
+    handle
+      .createCall(CALL_UPSERT_COURSE)
+      .bind(COURSE_NAME_PARAM, courseName)
+      .bind(COURSE_ACR_PARAM, courseAcr)
+      .bind(CALENDAR_SECTION_PARAM, calendarSection)
+      .bind(CALENDAR_TERM_PARAM, calendarTerm)
+      .bind(LANGUAGE_PARAM, language)
+      .bind(CATEGORY_PARAM, category)
+      .invoke()
+  }
+
+  override fun insertClassSectionEvent(
+    courseName: String?,
+    courseAcr: String?,
+    calendarSection: String?,
+    calendarTerm: String?,
+    summary: String,
+    description: String,
+    language: String,
+    category: String,
+    dtstart: Timestamp?,
+    dtend: Timestamp?,
+    weekDays: String?,
+    location: String?
+  ): Unit = tm.run(TransactionIsolationLevel.SERIALIZABLE) { handle ->
+    handle
+      .createCall(CALL_CREATE_EVENT)
+      .bind(COURSE_NAME_PARAM, courseName)
+      .bind(COURSE_ACR_PARAM, courseAcr)
+      .bind(CALENDAR_SECTION_PARAM, calendarSection)
+      .bind(CALENDAR_TERM_PARAM, calendarTerm)
+      .bind(SUMMARY_PARAM, summary)
+      .bind(DESCRIPTION_PARAM, description)
+      .bind(LANGUAGE_PARAM, language)
+      .bind(CATEGORY_PARAM, category)
+      .bind(DTSTART_PARAM, dtstart)
+      .bind(DTEND_PARAM, dtend)
+      .bind(WEEK_DAYS_PARAM, weekDays)
+      .bind(LOCATION_PARAM, location)
+      .invoke()
+  }
+
+  override fun transaction(t: (InsertClassSectionEventsRepo) -> Unit): Boolean {
+    tm.run { t(this) }
+    return true
+  }
```

Comment:
The function encapsulates all calls in a common transaction. This way, each and every change of the whole operation is guaranteed to rollback in case one of the Procedures fail.

Reply:
I'm stupid. Ignore this.